src/Controller/UserController.php

```diff
@@ -64,4 +78,64 @@ public function register(UserDto $userDto): Response
 
         return new JsonResponse($savedDto, Response::HTTP_CREATED);
     }
+
+    /**
+     * @Route(path="/delete/{email}", methods={"DELETE"})
+     */
+    public function softDelete(string $email): Response
+    {
+        $user = $this->userRepository->findOneBy(['email' => $email]);
+
+        if (null === $user) {
+            $this->logger->info('Soft delete failed: no account for such email');
+
+            return new JsonResponse('No account associated with email', Response::HTTP_NOT_FOUND, [], true);
+        }
+        $this->eventDispatcher->dispatch(new UserSoftDeleteEvent($user), UserSoftDeleteEvent::NAME);
+
+        $this->logger->info('Account soft deleted: ' . $email);
+
+        return new JsonResponse('Account deleted.', Response::HTTP_FOUND, [], true);
+    }
+
+    /**
+     * @Route(path="/recover/{email}", methods={"GET"})
+     */
+    public function softRecover(string $email): Response
+    {
+        $this->entityManager->getFilters()->disable('softdeleteable');
+        $recoveredUser = $this->userRepository->findOneBy(['email' => $email]);
+
+        if (null === $recoveredUser) {
+            $this->logger->info('Soft delete recover failed: no account for such email');
+
+            return new JsonResponse('No account associated with email', Response::HTTP_NOT_FOUND, [], true);
+        }
+
+        if (null === $recoveredUser->getDeletedAt()) {
+            $this->logger->info('Recover failed: the account is active: ' . $email);
+
+            return new JsonResponse('The account is already active!', Response::HTTP_OK, [], true);
+        }
+        $this->userRepository->recover($recoveredUser);
+
+        $this->logger->info('Account recovered when soft deleted: ' . $email);
+
+        return new JsonResponse('Account restored.', Response::HTTP_FOUND, [], true);
+    }
+
+    /**
+     * @Route(path="/show", methods={"GET"})
```

Comment:
path here is weird. you can just stick to `/api/users` for this to be more RESTful. Also, an endpoint which lists all users with no pagination is not a great idea. you might have at some point thousands of users